packages/headless/src/app/search-engine/search-engine.ts

```diff
@@ -111,20 +113,29 @@ export interface SearchEngineOptions
 export function buildSearchEngine(options: SearchEngineOptions): SearchEngine {
   const logger = buildLogger(options.loggerOptions);
   validateConfiguration(options.configuration, logger);
-
   const searchAPIClient = createSearchAPIClient(options.configuration, logger);
-  const generatedAnswerClient = createGeneratedAnswerAPIClient(logger);
 
   const thunkArguments = {
     ...buildThunkExtraArguments(options.configuration, logger),
     apiClient: searchAPIClient,
-    streamingClient: generatedAnswerClient,
+    ...(options.configuration.answerConfigurationId
+      ? {}
+      : {streamingClient: createLegacyGeneratedAnswerAPIClient(logger)}),
   };
 
-  const augmentedOptions: EngineOptions<SearchEngineReducers> = {
+  type EngineOptionsWithAnswerMiddlewares = Omit<
+    EngineOptions<SearchEngineReducers>,
+    'middlewares'
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  > & {middlewares?: any};
```

Comment:
This is not a lot better than casting the middlewares type at line 137. But I did not find any better way to pass the answerApi.middleware in the options.